packages/runtime/src/lib/parsing.ts

```diff
@@ -21,6 +21,26 @@ export function assertOnlyDigits (val: string, line: number): void {
   }
 }
 
+/**
+ * assert if string is valid number respesent.
+ * return val if format is correct
+ * @param val : string
+ */
+export function assertNumber (val: string, line: number): string {
+  if (reOct.test(val)) {
+    // typescript use 0o postfix instade of 0 postfix for oct format.
+    return "0o".concat(val.substring(1));
+  }
+
+  if (reDec.test(val) || reHex.test(val)) return val;
```

Comment:
before returning, should we try to parse the number and check if it's smaller than `1<<64` (2**64)?

Reply:
We check it in Int opcode [here](https://github.com/scale-it/algo-builder/blob/dc544299516315c33ce0269d01fcbe18decbc68f/packages/runtime/src/interpreter/opcode-list.ts#L2154)